python/cudf/cudf/core/multiindex.py

```diff
@@ -123,7 +123,7 @@ def names(self):
     def names(self, value):
         value = [None] * self.nlevels if value is None else value
         assert len(value) == self.nlevels
-        self._names = value
+        self._names = [n for n in value]
```

Comment:
I think it's a bad assumption to assume `value` here can be efficiently iterated over. Maybe just call `list(value)`?